compiler/rustc_mir/src/transform/const_prop.rs

```diff
@@ -416,11 +421,27 @@ impl<'mir, 'tcx> ConstPropagator<'mir, 'tcx> {
         }
     }
 
+    /// Some `InterpError`s could be ignored but must not be to ensure that queries are stable.
+    fn check_interpresult<T>(tcx: TyCtxt<'tcx>, error: &InterpResult<'tcx, T>) {
```

Comment:
Uff, okay, this function is called a lot and I have no idea if this makes any sense. :/ Also it duplicates the `matches!` with the code in `eval_constant`.